smart/model.py

```diff
@@ -2199,3 +2264,40 @@ def adjust_dt(self):
         #     dt_scale = min(dt_scale * 0.8, 0.8)
         dt_cur = float(self.dt) * dt_scale
         self.set_dt(dt_cur)
+
+    def load_vector(self, h5_file, tVec):
+        cur_level = d.get_log_level()
+        d.set_log_level(40)  # suppress warning about rank of h5 data
+        with d.HDF5File(self.parent_mesh.mpi_comm, h5_file, "r") as cur_file:
+            # Find index associated with the starting time
+            if np.any(np.isclose(tVec, float(self.t))):
+                vec_new = d.Vector()
+                idx1 = np.nonzero(np.isclose(tVec, float(self.t)))[0][0]
```

Comment:
```suggestion
                idx1 = np.flatnonzero(np.isclose(tVec, float(self.t)))[0]
```
I would probably only call this once outside the if, ie
```
has_timestamp = np.flatnonzero(np.isclose(tVec, float(self.t)))
if len(has_timestamp)>0:
    idx1 = has_timestamp[0]
```